src/finch/tensor.py

```diff
@@ -587,25 +615,38 @@ def asarray(obj, /, *, dtype=None, format=None):
         return tensor
 
 
+def reshape(x: Tensor, /, shape: tuple[int, ...], *, copy: bool | None = None) -> Tensor:
+    # TODO: https://github.com/willow-ahrens/Finch.jl/issues/558
+    #       Only to run array-api-tests that require it for multiple tests.
+    #       Must be reimplemented once `reshape` is available in Finch.jl.
+    arr = x.todense()
```

Comment:
Could we simply add a PerformanceWarning here that we are densifying?

Reply:
Done!